packages/integration-karma/test/shadow-dom/multiple-templates/index.spec.js

```diff
@@ -0,0 +1,41 @@
+import { createElement } from 'lwc';
+import Multi from 'x/multi';
+
+if (process.env.NATIVE_SHADOW) {
+    describe('Shadow DOM styling - multiple shadow DOM components', () => {
+        it('Does not duplicate styles if template is re-rendered', () => {
```

Comment:
This test confirms we don't end up with duplicate `<style>`s/`CSSStyleSheet`s in the shadow root when we render template A, then B, then A again, etc.

Arguably we could just not care about this, and then the logic would be simpler because we wouldn't have to check for existing styles. I could go either way, but I kind of like that we don't endlessly add `<style>`s if (for whatever reason) the component is alternating between A and B repeatedly.